### ISSUE TITLE
Need contribution and local environment setup guide.

### ISSUE THREAD

**Issue by amit-bhandari** [cooperhammond/irs#46]
I want to make some changes in the irs locally, but have no idea how to build system after my changes.

**Comment by dylanbaxter** [comment]
+1
I find this little app very handy but had to make some changes to installed code to get it running.

I had received an error from Selenium about chromedriver not being in my PATH, even though it totally was in my path (I had symlink'ed it to /usr/bin/chromedriver, however). By removing the line specifying the chromedriver path, Selenium just runs chromebrowser assuming it's actually in the PATH, which should eliminate some confusion for others.
Once around that little hurdle, I got another error about some GTK-type library failing to load, which given that I'm running IRS on a headless system didn't make sense. In addition, since IRS is a CLI app, why should chromebrowser load all the extra UI garbage? So passing the --headless option to chromebrowser was the solution there.

I ended up with the following changes:

 /usr/local/lib/python2.7/dist-packages/irs/utils.py, line 467:
``` python
        executable_path = {'executable_path': '/usr/local/bin/chromedriver'}
        ...
        with Browser('chrome', **executable_path) as b:
```
Changed to:
``` python
        chrome_options = webdriver.ChromeOptions()
        chrome_options.add_argument('--headless')
        with Browser('chrome', options=chrome_options) as b:
```
Of course an import statement for webdriver was added above somewhere:
``` python
from selenium import webdriver
```

Now that I have IRS running, I have run into a few 403 errors that break program execution. I would like to wrap the youtube_dl download call in a try/catch to make the attempt a certain number of tries before exiting cleanly with a status code.

That would take place in the following location:
/usr/local/lib/python2.7/dist-packages/irs/ripper.py, line 453:
``` python
        with youtube_dl.YoutubeDL(ydl_opts) as ydl:
            ydl.download([video_url])
```

I'm probably gonna work on that today...
And along those lines, I also want to script this process to download a list of albums unattended and would need cleaner control and exit management in general to help with that.

The bottom line for me is that I am willing to contribute some work if the author would have it, and perhaps help me to understand what my development environment should look like.

**Comment by cooperhammond** [comment]
I'm definitely willing to merge any reasonable pull requests. I just released a major revamp of the program. Sorry it has been so slow, school has been busy.
